src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.model.RoleName;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getByName(RoleName roleName) {
+        try (Session session = factory.openSession()) {
+            return session.createQuery("FROM Role r "
+                            + "WHERE r.roleName LIKE :roleName", Role.class)
```

Comment:
```suggestion
                            + "WHERE r.roleName = :roleName", Role.class)
```
what is the difference between `=` and `LIKE`?

Reply:
`=` is used to find equal value and `LIKE` is used to find similar to some pattern value